R/unnest.R

```diff
@@ -157,7 +157,20 @@ unnest.data.frame <- function(data, ..., .drop = NA, .id = NULL,
   rest <- data[rep(seq_nrow(data), n[[1]]), group_vars, drop = FALSE]
   out <- dplyr::bind_cols(rest, unnested_atomic, unnested_dataframe)
 
-  reconstruct_tibble(data, out)
+  # Restore column order
+  nested_vars <- imap(data, function(.x, .y) {
+    if (is_empty(.x)) {
+      return(.y)
+    } else if (is.data.frame(.x[[1]])) {
+      return(names(.x[[1]]))
+    } else {
+      return(.y)
+    }
+  })
```

Comment:
I feel like this approach is easier to follow. Thoughts?

```r
  nested_vars <- imap(data, function(.x, .y) {
    if (is_empty(.x)) {
      return(.y)
    }
    .x <- .x[[1]]
    if (is.data.frame(.x)) {
      names(.x)
    } else {
      .y
    }
  })
```

Reply:
Not at all! You gotta do what's best for the package and I'll just be happy to see the issue fixed. Either way, I'm always happy to help out where I can.